src/support/modify.c

```diff
@@ -38,6 +38,33 @@
     }                         \
     while (0)
 
+/*
+ * __wt_modify_idempotent --
+ *     Check if a modify operation is idempotent.
+ */
+bool
+__wt_modify_idempotent(const void *modify)
+{
+    WT_MODIFY mod;
+    size_t tmp;
+    const size_t *p;
+    int nentries;
+
+    /* Get the number of modify entries. */
+    p = modify;
+    memcpy(&tmp, p++, sizeof(size_t));
+    nentries = (int)tmp;
+
+    WT_MODIFY_FOREACH_BEGIN(mod, p, nentries, 0)
```

Comment:
Nit: Not introduced by your change but could you please add `WT_MODIFY_FOREACH_BEGIN` to the `.clang-format` file as a loop macro? I must have missed this one.

Reply:
Done.  Also sorted the list while I was in the area.